src/jit/lsraxarch.cpp

```diff
@@ -2378,6 +2384,38 @@ void LinearScan::TreeNodeInfoInitSIMD(GenTreeSIMD* simdTree)
 }
 #endif // FEATURE_SIMD
 
+#if defined(_TARGET_XARCH_) && !defined(LEGACY_BACKEND)
+//------------------------------------------------------------------------
+// TreeNodeInfoInitHWIntrinsic: Set the NodeInfo for a GT_HWIntrinsic tree.
+//
+// Arguments:
+//    tree       - The GT_HWIntrinsic node of interest
+//
+// Return Value:
+//    None.
+
+void LinearScan::TreeNodeInfoInitHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
+{
+    TreeNodeInfo* info = &(intrinsicTree->gtLsraInfo);
+    switch (intrinsicTree->gtHWIntrinsicId)
```

Comment:
For these, and presumably for most of the to-be-supported intrinsics, since they should directly map to single instructions, this could presumably just be:
```
if (tree->gtGetOp2IfPresent() != nullptr)
{
    info->srcCount += GetOperandSourceCount(tree->gtOp.gtOp2);
}
info->srcCount += GetOperandSourceCount(tree->gtOp.gtOp1);
```
And you could optionally add an assert that the intrinsic id is one of the expected values.

Reply:
Good point!